### PR TITLE
Chapter 8

### PR DESCRIPTION
**Pull request by mirulzaei**

### CODE REVIEW

**Comment by mirulzaei** [comment]
ready

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -13,7 +13,6 @@
...
  *= require_self
  */
- @import "bootstrap-sprockets";
- @import "bootstrap";
- @import "color";
- 
\ No newline at end of file
+ @import 'bootstrap-sprockets';
+ @import 'bootstrap';
+ @import 'color';
```
Comment (saobv-2223) [inline app/assets/stylesheets/application.scss]:
```suggestion
 @import 'color';
 
```

File: app/assets/stylesheets/sessions.scss [hunk 2]
```diff
@@ -1,3 +1,3 @@
-// Place all the styles related to the StaticPages controller here.
+// Place all the styles related to the Sessions controller here.
 // They will automatically be included in application.css.
 // You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/sessions.scss]: remove file that never used

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -2,5 +2,5 @@ module ApplicationHelper
   def full_title page_title
     base_title = "Sample App"
     page_title.blank? ? base_title : "#{page_title} | #{base_title}"
-  end  
+
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]:
```suggestion
  end
```

File: app/helpers/sessions_helper.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
+  end
+
+  def logged_in?
+    current_user.present?
+  end
+
+  def log_out
+    session.delete(:user_id)
```
Comment (saobv-2223) [inline app/helpers/sessions_helper.rb]: remove ()

File: app/javascript/packs/application.js [hunk 5]
```diff
@@ -3,5 +3,6 @@ require('@rails/ujs').start()
 require('turbolinks').start()
 require('@rails/activestorage').start()
 require('channels')
-
+require("jquery")
+import "bootstrap"
```
Comment (saobv-2223) [inline app/javascript/packs/application.js]: using '

File: app/models/user.rb [hunk 6]
```diff
@@ -15,9 +15,20 @@ class User < ApplicationRecord
...
+    BCrypt::Password.create string, cost: cost
+  end
+
   private
 
   def downcase_email
     email.downcase!
   end
+
+
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
```

File: app/views/sessions/new.html.erb [hunk 7]
```diff
@@ -1,2 +1,14 @@
...
+    <%= form_for :session, url: login_path do |f| %>
+      <%= f.label :email %>
+      <%= f.email_field :email, class: "form-control" %>
+      <%= f.label :password %>
+      <%= f.password_field :password, class: "form-control" %>
+      <%= f.submit "Log in", class: "btn btn-primary" %>
+    <% end %>
+    <p><%= t "new_user?" %><%= link_to t("sign_up_now!"), signup_path %></p>
+  </div>
+</div>
```
Comment (saobv-2223) [inline app/views/sessions/new.html.erb]:
```suggestion
</div>

```

File: config/routes.rb [hunk 8]
```diff
@@ -1,12 +1,13 @@
...
-  
   get "/home", to: "static_pages#home"
   get "/help", to: "static_pages#help"
   get "/about", to: "static_pages#about"
   get "/contact", to: "static_pages#contact"
   get "/signup", to: "users#new"
+  get "/login",to: "sessions#new"
+  post "/login", to: "sessions#create"
+  delete "/logout", to: "sessions#destroy"
+
```
Comment (saobv-2223) [inline config/routes.rb]:
```suggestion
  delete "/logout", to: "sessions#destroy"
```

**Comment by mirulzaei** [comment]
ready

File: app/assets/stylesheets/application.scss [hunk 9]
```diff
@@ -13,7 +13,7 @@
  *= require_tree .
  *= require_self
  */
- @import "bootstrap-sprockets";
- @import "bootstrap";
- @import "color";
+ @import 'bootstrap-sprockets';
+ @import 'bootstrap';
+ @import 'color';
```
Comment (saobv-2223) [inline app/assets/stylesheets/application.scss:18]:
```suggestion
 @import 'color';
 
```

**Review by saobv-2223 — COMMENTED** [review]
fix comment and merge

**Pull request merged**